packages/fhir-router/src/repo.ts

```diff
@@ -92,6 +94,19 @@ export interface FhirRepository<TClient = unknown> {
    */
   updateResource<T extends Resource>(resource: T, versionId?: string): Promise<T>;
 
+  /**
+   * Creates or updates a resource based on a search criteria.
+   *
+   * See: https://www.hl7.org/fhir/http.html#cond-update
+   * @param resource - The FHIR resource to create or update.
+   * @param search - The search criteria to find an existing resource to update.
+   * @returns The updated resource.
+   */
+  conditionalUpdate<T extends Resource>(
```

Comment:
Hm, all resource-oriented operations currently include `Resource` in the function name.

`conditionallyUpdateResource` is probably the most technically accurate but long and an eyesore.

Objections to `upsertResource()`, which would then align with the name in `MedplumClient`?